noise.js

```diff
@@ -124,12 +125,17 @@ module.exports = class NoiseState extends SymmetricState {
   }
 
   final () {
-    const { cipher1, cipher2 } = this.split()
+    const [k1, k2] = this.split()
 
-    this.tx = this.initiator ? cipher1 : cipher2
-    this.rx = this.initiator ? cipher2 : cipher1
+    this.tx = this.initiator ? k1 : k2
+    this.rx = this.initiator ? k2 : k1
 
     this.handshakeComplete = true
+    this.handshakeHash = this.getHandshakeHash()
+
+    this.remotePublicKey = new Uint8Array(this.rs)
```

Comment:
could we set the immediately when we set .rs ? (we can also just use .rs as the name if you prefer)